Eggstravaganza/Assets/Scripts/EggSpawn.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+[Serializable]
+public class Egg
+{
+    public GameObject prefab;
+    [Range(0f, 100)]
+    public float chance = 100;
+    [HideInInspector]
+    public double weight;
+}
+
+public class EggSpawn : MonoBehaviour
+{
+    public List<Egg> eggs;
+    
+    double m_AccumulatedWeights;
+    System.Random m_Rand = new();
+
+    void Awake()
+    {
+        CalculateWeights();
+    }
+
+    public void SpawnEgg()
+    {
+        var currentEggToSpawn = eggs[ChooseEggToSpawn()];
+        
+        // TODO: Maybe modify spawn height? Not sure what value we'd want yet
+        var randomPosition = Random.insideUnitCircle;
```

Comment:
maybe we could also add an exposed property for setting the scale of the spawn area? maybe a vec2 that you multiply `randomPosition` by?